Modules/Loadable/Segmentations/EditorEffects/qSlicerSegmentEditorPaintEffect.cxx

```diff
@@ -153,6 +154,15 @@ class BrushPipeline2D : public BrushPipeline
     feedbackActorProperty->SetOpacity(0.5);
     this->FeedbackActor->SetMapper(this->FeedbackMapper);
     this->FeedbackActor->VisibilityOff();
+
+    // If there is no input or if the input has no points, the vtkTransformPolyDataFilter will display an error message
+    // on every update: "No input data", polluting the log and slowing down the application.
+    // To prevent logging the error, an error callback command is set that discards the message.
+    // Ideally, vtkTransformPolyDataFilter should be updated to not log this unnecessary error message.
```

Comment:
maybe make an upstream merge request and then cherry pick it into our vtk fork?

Reply:
Yes, of course, that would be better. We have 3 other workarounds because of this VTK error (search for "No input data" in the Slicer source code). I'll try to put together a pull reuest that removes the message in VTK.

Reply:
It looks like VTK is pretty inconsistent, but probably all of these should be `vtkDebugMacro`

https://gitlab.kitware.com/search?search=No%20input%20data&nav_source=navbar&project_id=13&group_id=16&search_code=true&repository_ref=master

Reply:
VTK just logs this error in 23 classes, out of a total of about 3000 classes. In vtkTransformPolyDataFilter the message was added when the filter was originally developed in 1995. The other occurrences may be just copy-paste or time saving for the developer (the developer decided to just return an error and let the user deal with the situation instead of making sure the class works well with empty input). I've created a merge request in VTK to remove the error message (the filter seems to work well for empty input):

https://gitlab.kitware.com/vtk/vtk/-/merge_requests/10472